r/configure

```diff
@@ -91,6 +96,14 @@ else
       fi
       PKG_CFLAGS="-I$BREWDIR/opt/$PKG_BREW_NAME/include"
       PKG_LIBS="-L$BREWDIR/opt/$PKG_BREW_NAME/lib $PKG_LIBS"
+    else
+      VERSION=$(grep ^Version DESCRIPTION | sed s/Version:\ //)
+      ${R_HOME}/bin/Rscript tools/linuxlibs.R $VERSION $LOCAL_LIBARROW
+      PKG_CFLAGS="-I$(pwd)/libarrow/arrow-${VERSION}/include $PKG_CFLAGS"
+      PKG_LIBS="-L$(pwd)/libarrow/arrow-${VERSION}/lib $PKG_LIBS"
+      # Also enumerate the static libs included in there
+      # TODO: this should be generated based on what's in the lib dir
+      PKG_LIBS="$PKG_LIBS -lthrift -lsnappy -lz -lzstd -llz4 -lbrotlidec-static -lbrotlienc-static -lbrotlicommon-static -lboost_filesystem -lboost_regex -lboost_system -ljemalloc_pic"
```

Comment:
How about using only static linked `libarrow.so` (shared library) instead of using `libarrow.a` (static library) and other static libraries?

Reply:
My understanding (from e.g. https://github.com/r-windows/rtools-packages#building-package and the similar macOS system) is that we can only download static libs during package installation. The package gets built in a temp directory and creates a shared library `arrow.so` from the R C++ binding code and these libraries, so if we downloaded and linked a shared `libarrow.so` in that, the package won't be able to load because it won't be able to find `libarrow.so` at runtime.

Reply:
👍 see also https://github.com/apache/arrow/pull/6073 and discussion.

Reply:
Then you can definitively use rpath with `$ORIGIN`.

Reply:
But I'm not building a shared library, so rpath is irrelevant, right?